dot/types/babe.go

```diff
@@ -108,6 +109,10 @@ type ConfigData struct {
 
 // GetSlotFromHeader returns the BABE slot from the given header
 func GetSlotFromHeader(header *Header) (uint64, error) {
+	if header.Number == 0 {
+		return 0, ErrGenesisHeader
```

Comment:
I don't really care much for this, tbh. But if you want to start wrapping errors, instead of doing this here, we can just create our errors with `fmt.Errorf` instead of `errors.New`. 

@timwu20 what do you think about this change?

Reply:
> But if you want to start wrapping errors, instead of doing this here, we can just create our errors with fmt.Errorf instead of errors.New

Only changing `errors.New` for `fmt.Errorf` we don't wrap errors, to wrap an error we should explicity use `fmt.Errorf("%w", err)`. 

The suggestion is to enforce the usage of `errors.Is` and changing`errors.New` for `fmt.Errorf` we keep the problem (allowing error assertion with `==` instead of `errors.Is`)

Reply:
What are we discussing here? Whether or not we define errors via `errors.New` or `fmt.Errorf`?

Reply:
@EclesioMeloJunior my bad, why do you want to force `errors.Is`?

Reply:
@kishansagathiya I am not advocating for enforcing `errors.Is`, I commented that using `fmt.Errorf` to create errors instead of `errors.New` (as you seggested in https://github.com/ChainSafe/gossamer/pull/2897) is not the same as wrapping an error and is really odd to me.

And regards to the @qdm12 suggestion, I tested in the [go.dev](https://go.dev/play/p/oXGHNROAgm6) if using `fmt.Errorf` to create errors enforces the usage of `errors.Is` over the `==` and it does not enforce.

Reply:
pretty sure `errors.Is` works in this case on the calling side if testing for `ErrGenesisHeader`.  I'm not sure why we need to wrap this error when returning as based on the suggestion.

Reply:
@qdm12 does not look like I need to do anything here. Please un-resolve if you think otherwise.

Reply:
Oh well this really was a simple suggestion and the conversation really went overboard :smile: 

The reason for `errors.Is` and 'disallowing' `==` is to allow later to add context without breaking compatibility with callers.

From https://go.dev/blog/go1.13-errors

> If a function is defined as returning an error wrapping some sentinel or type, do not return the underlying error directly.

```go
...
return fmt.Errorf("%w", ErrPermission)
...
```

Reply:
I don't really agree with the go blog in this case.  Like I see how this would enforce callers of this function to use `errors.Is`, but we are copying an error and not adding context and passing that around.  We can just enforce the use of `errors.Is` through PRs and mutual understanding.

Reply:
think, it is safe to resolve this.